Bugzilla/CGI.pm

```diff
@@ -34,9 +34,9 @@ BEGIN {
 sub DEFAULT_CSP {
     my %policy = (
         default_src => [ 'self' ],
-        script_src  => [ 'self', 'unsafe-inline', 'unsafe-eval' ],
+        script_src  => [ 'self', 'unsafe-inline', 'unsafe-eval', 'https://www.google-analytics.com' ],
         child_src   => [ 'self', ],
-        img_src     => [ 'self', 'https://secure.gravatar.com' ],
+        img_src     => [ 'self', 'https://secure.gravatar.com', 'https://www.google-analytics.com' ],
```

Comment:
I highly recommend that you use XHR over images for GA.

See https://githubengineering.com/githubs-post-csp-journey/ for specifics as to the security risks of images vs. XHR.

Reply:
Umm, but this is what (all?) other Mozilla properties are doing.

Reply:
Also, because BMO stores/serves attachments on different hosts, the security risk could be mitigated, I guess.

Reply:
Reed is correct, but we're already vulnerable to this with gravatar 's d=parameter. We should file a *separate* bug to fix both (lock-down img-src to only urls we control and use XHR for GA and an image proxy for gravatar)

Reply:
Just for reference: BzDeck was using a [service worker](https://github.com/bzdeck/bzdeck/blob/master/webroot/static/scripts/workers/service-worker.js) to proxy Gravatar requests.

Reply:
@kyoshino does that mean the origin of a service worker is different than a direct <img>? do we need img-src in that case? Regardless, this is off-topic -- I'll file a new bug for this deficiency

Reply:
BzDeck's [CSP](https://github.com/bzdeck/bzdeck/blob/master/webroot/.htaccess#L60-L63) still had `https://secure.gravatar.com` for `img-src`. I thought that was necessary for some reason.